### DISCUSSION TITLE
Useless genes make the entire genetics system less fun

### DISCUSSION THREAD

**Discussion by Vykn** [FakeFishGames/Barotrauma#12902] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

This issue is less unintentional, more so oversight.

There are 12 types of genetic materials - Crawler, hammerhead, mantis, moloch, thresher, matriarch, hunter, mollusc, skitter, husk, mudraptor & spineling. Of note for this particular issue are the skitter, mollusc, mudraptor, and spineling types.

These genes are notorious for being just plain useless. Mudraptor and Spineling look interesting, but serve absolutely zero practical purpose. I think these are less of an issue with these types because they're fashionable and are a little fun, but they do affect the genetics system as a whole.
Mollusc and Skitter may sound useful when reading the description, but actually checking how effective they are shows how utterly terrible they are. The strongest Mollusc gene grants +10 vigor upon being hit. That is a 0.83% damage resistance. Not 83%, but *0.83%*. A rounding error. Stacking it by being hit multiple times is also not practical in the slightest - see figure 1 below. Skitter behaves in exactly the same way.

https://github.com/Regalis11/Barotrauma/assets/37850646/16b23232-dc9b-41c4-b482-cb08918420a8

[Figure 1. a character with 100% Mollusc and Skitter genes being torn apart by a dozen crawlers, and receiving a pittance of vigor and hyperactivity]

The fact that these genes aren't useful isn't a major problem in a vacuum - it's not like they're negative effects - but rather the issue is that these genes share the UGM loot pool with all the other more desirable genes. 4 out of the 12 genes that are in the game serve zero practical purpose. Every time a sample of UGM is researched, there's a 1 in 3 chance that the output is just useless. This makes obtaining high quality desirable genes more of a grind, and can lead to significant disappointment when researching if players get unlucky. If these useless genes did not exist, then the useful genes would be more common, and so the existence of these useless genes almost exclusively serve to make the whole genetics system less fun to engage with.

The options to address this issue that I see:
1. Remove these useless genes
2. Rework these genes to be not useless
3. Rework the entire genetics system

Option 1 would certainly efficiently address the problem, though some people do like the Spineling and Mudraptor genes because they're one of the rare ways to make one's character look unique, and removing them might come as a disappointment.

Option 2 would highly depend on how they are reworked - ideally they should be comparable in practicality as the other genes -- see @ubertpendragon 's ~~mod~~ PR #12383 as a possible example. I think the points raised are very good - reworked genes ought to be useful, and actually reward refining to higher quality genes. 
I think reworking Mollusc would be a little trickier for the same reason that Hunter genes are so incredibly good. These genes take effect when getting hit by an enemy - for Hunter, this means that whenever a creature is close enough to hit you, you get 2.5x melee attack power. In melee, you are expected to be hit, and so the "when you are hit" clause is basically redundant. If Mollusc was reworked like Hunter is implemented, it might just become a better Moloch gene. I think reworking Skitter like Hunter would work reasonably well - a character gains an appreciable increase in movement speed when attacked - useful, unique, and probably what the original design for Skitter was supposed to be.
Alternatively, some gene types could be made less common than others, though that might be overcomplicating things.

Option 3 could be literally anything. There are many players with ideas on how to change things, and while I think that the genetics system is quite fine as-is (despite being incredibly opaque and confusing), it hasn't really been touched at all since it was implemented.

More could potentially be said about the overall balance of genetics, but this is what I see as the biggest issue of an overall very solid and fun game mechanic.

### Reproduction steps

1. Use skitter/mollusc genes
2. be hit by many attacks
3. observe that the effect from these genes is negligible

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.0.21.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Vykn** [comment]
Addendum 1: Thresher genes

Thresher isn't as bad as Mollusc & Skitter, but still has notable issues;

- **Essentially useless until 100%** Getting to 100% gene quality is very laborious, and below 100% Thresher only delays suffocation, which isn't particularly useful. It doesn't even work while the character is unconscious, and it doesn't reduce oxygen tank consumption.

- **Not even worth using at 100%** Wearing suits without an oxygen tank *always* causes beeping, which is exceedingly annoying and discourages players from using the genes as they were intended. See #12143.

There is basically no use case for Thresher genes. Oxygen is never an issue, but if that is being addressed, and Thresher genes are fixed, then they could be good genes.

**Comment by Regalis11** [comment]
Addressed Mollusk and Skitter genes in https://github.com/Regalis11/Barotrauma-development/commit/a97c294f3218dd25cd9d5780d9af550a99751d45. The rest need to still be looked into. Note for QA: if this gets through review and QA before the remaining genes have been addressed, just bounce the ticket back to ToDo.

**Comment by Jasontti** [comment]
Tested skitter and mollusc genes.
No issues found.

**Comment by Regalis11** [comment]
@Jasontti How did you test them? Any thoughts on how well the new durations work, do they need some adjustments still?

**Comment by Jasontti** [comment]
@Regalis11 
Buff duration on both of them is on the long side ~10 minutes, but on the other hand it's defined in the affliction itself and other genes have infinite duration so i would not change this.
As for usefulness, it makes a huge difference for example in mechanics melee build. You can 2 hit kill crawlers with heavy wrench.

One issue that i noticed now when testing again is, that afflictions go straight to "heavy" after receiving one hit with sufficient damage, even from other crew members. Maybe <requiredaffliction type="damage" minstrength="5" /> could be tiered more for different stages of the affliction, up to 50, so that more you get damaged stronger you get.

**Comment by Regalis11** [comment]
Could not reproduce. The mollusc and skitter genes seem to give a maximum of 10 units of vigor/haste when the character is attacked once, which actually seems way too low?

**Comment by Jasontti** [comment]
Repro steps:
1. MP with 2 players, one must have genesplicer and mollusc/skitter genes.
2. Hit gened up player with a wrench. No genetic activation and you can hit as many times as you want.
3. Heal player.
4. Hit same player with crowbar.
5. Skitter and mollusc genes activate straight to 600 vigor and hyperactivity. (Sometimes it's only 550 vigor)

I might be misunderstanding this as that this affliction needs to be on character to apply strengthen.
AfflictionsGeneticMaterial.xml
`<requiredaffliction type="damage" minstrength="5" />`

**Comment by Regalis11** [comment]
Looks like I was testing this in an incorrect branch. 🤦‍♂️ Addressed in https://github.com/Regalis11/Barotrauma-development/commit/c7bd3c123a115d3678a047cbdd035f2a1d4d93c3

**Comment by Jasontti** [comment]
Tested.
No issues found.

**Comment by itchyOwl** [comment]
So this issue is fixed? Why wasn't the ticket closed? Was there still something to do?

**Comment by Jasontti** [comment]
No issues, must have been a mistake.

**Comment by Regalis11** [comment]
> Note for QA: if this gets through review and QA before the remaining genes have been addressed, just bounce the ticket back to ToDo.

Only the bugged mollusc and skitter genes have been fixed, but we haven't done anything to the spineling and mudraptor genes.
